Detectors/MUON/MID/Simulation/src/DigitsMerger.cxx

```diff
@@ -33,32 +33,41 @@ void DigitsMerger::mergeDigit(size_t idigit, const std::vector<ColumnDataMC>& in
   mDigitsLabels.emplace_back(std::make_pair(inDigitStore[idigit], vec));
 }
 
-void DigitsMerger::process(const std::vector<ColumnDataMC>& inDigitStore, const o2::dataformats::MCTruthContainer<MCLabel>& inMCContainer, const std::vector<ROFRecord>& inROFRecords)
+void DigitsMerger::process(const std::vector<ColumnDataMC>& inDigitStore, const o2::dataformats::MCTruthContainer<MCLabel>& inMCContainer, const std::vector<ROFRecord>& inROFRecords, bool mergeInBunchPileup)
 {
   /// Merges the MC digits that are provided per hit
   /// into the format that we expect from data
   /// \param inDigitStore Vector of input MC digits
   /// \param inMCContainer Container with MC labels for input MC digits
   /// \param inROFRecords Vector with RO frame records
+  /// \param mergeInBunchPileup Merge the digits coming from in-bunch pileup
   mDigitStore.clear();
   mMCContainer.clear();
   mROFRecords.clear();
+  mDigitsLabels.clear();
 
-  for (auto& rofRecord : inROFRecords) {
-    mDigitsLabels.clear();
-    for (size_t idigit = rofRecord.firstEntry; idigit < rofRecord.firstEntry + rofRecord.nEntries; ++idigit) {
+  for (auto rofIt = inROFRecords.begin(); rofIt != inROFRecords.end(); ++rofIt) {
+    auto nextRofIt = rofIt + 1;
+    bool mergeInteractions = mergeInBunchPileup && nextRofIt != inROFRecords.end() && rofIt->interactionRecord.toLong() == nextRofIt->interactionRecord.toLong();
```

Comment:
this is correct, just for the record: InteractionRecord has full set of ``==, <, >`` operators, no need to use toLong.

Reply:
You're right! I changed it, since in this way it is much cleaner. Thanks!